### PR TITLE
feat: add fragments option to the scanner itself

### PR DESCRIPTION
**Pull request by wjones127**
To support setting other scanner options along with the fragments, add `fragments` as an argument in the scanner itself.

### CODE REVIEW

File: python/python/lance/dataset.py [hunk 1]
```diff
@@ -587,6 +595,19 @@ def filter(self, filter: Union[str, pa.compute.Expression]) -> ScannerBuilder:
...
         return self
 
+    def with_fragments(self, fragments: Optional[Iterable[LanceFragment]]) -> ScannerBuilder:
+        if fragments is not None:
+            inner_fragments = []
+            for f in fragments:
+                if isinstance(f, LanceFragment):
+                    inner_fragments.append(f._fragment)
+                else:
+                    raise TypeError("fragments must be an iterable of LanceFragment")
```
Comment (eddyxu) [inline python/python/lance/dataset.py:605]: Could you also add a message to tell user what the current type of `f` is , for easy debugging?

File: python/python/lance/dataset.py [hunk 2]
```diff
@@ -687,16 +709,18 @@ def from_fragments(fragments: Iterable[LanceFragment], **kwargs):
         -------
         LanceScanner
         """
-        # TODO: can we take kwargs here for the scanner builder?
-        # TODO: peek instead of list
+        warnings.warn(
+            "Deprecated in version 0.4.13. Use the fragments argument in LanceDataset.scanner() instead.",
```
Comment (eddyxu) [inline python/python/lance/dataset.py:713]: we might only have 1-2 users are using this method.  Let's make sure they know, and we can delete this method in a few versions later?

Comment (eddyxu) [reply python/python/lance/dataset.py:713]: Actually this method is `semi-private` method, right? so users dont directly use it, instead they use `Dataset.scanner(fragments=...)`?

Comment (wjones127) [reply python/python/lance/dataset.py:713]: `Dataset.scanner(fragments=...)` doesn't use this function, so if no one is using `from_fragments()` we can delete it.

Comment (eddyxu) [reply python/python/lance/dataset.py:713]: Sg. Let's delete it then.

**Review by eddyxu — APPROVED** [review]

**Pull request merged**